### PR TITLE
klogr call depth

### PR DESCRIPTION
**Pull request by pohly**
**What this PR does / why we need it**:

Exposing the call depth support from klog also through the logr interface makes it possible to write generic helper functions around logr instances with correct call locations in the final log messages.

**Special notes for your reviewer**:

Based on https://github.com/kubernetes/klog/pull/197

**Release note**:
```release-note
implement logr.WithCallDepth
```

### CODE REVIEW

File: klogr/klogr.go [hunk 1]
```diff
@@ -255,4 +260,11 @@ func (l klogger) WithValues(kvList ...interface{}) logr.Logger {
...
 }
 
+func (l klogger) WithCallDepth(depth int) logr.Logger {
+	new := l.clone()
+	new.callDepth += depth
+	return new
+}
+
 var _ logr.Logger = klogger{}
+var _ logr.CallDepthLogger = klogger{}
```
Comment (pohly) [inline klogr/klogr.go:270]: Usage of the `CallDepthLogger` type in this compile-time check is the only reason why this PR updates logr to master.

We can either:
- wait for a new tagged logr release and use that or
- omit this check here and leave the logr dependency unchanged.

**Review by neolit123 — COMMENTED** [review]
the InfoSDepth / ErrorSDepth additions seem fine to me and i'm not seeing an API breaking change.

maybe this should get review by the contributors doing the struct. logging extensions to klog.

**Comment by dims** [comment]
@pohly thanks to @thockin we have a new tag https://github.com/go-logr/logr/releases/tag/v0.4.0 please go ahead and update this PR

**Comment by pohly** [comment]
> please go ahead and update this PR

Done. It also passes all tests now (still needed to adapt a bit to the new API in one test).

**Comment by dims** [comment]
/approve
/lgtm

@pohly please feel free to submit a follow PR for the test update you mentioned.

**Comment by pohly** [comment]
> please feel free to submit a follow PR for the test update you mentioned.

My wording was misleading: that change is done and included in the updated PR.

Can we do a new klog release with this change?

**Pull request merged**